autogpt_platform/backend/schema.prisma

```diff
@@ -19,45 +19,148 @@ model User {
   updatedAt DateTime @updatedAt
 
   // Relations
-  AgentGraphs                  AgentGraph[]
+  AgentGraphs                  Agent[]
   AgentGraphExecutions         AgentGraphExecution[]
   AgentGraphExecutionSchedules AgentGraphExecutionSchedule[]
   AnalyticsDetails             AnalyticsDetails[]
   AnalyticsMetrics             AnalyticsMetrics[]
   UserBlockCredit              UserBlockCredit[]
+  ConfiguredAgents             ConfiguredAgent[]
+  UserAgents                   UserAgents[]
+
+  // User Group relations
+  groups                 UserGroup[]              @relation("GroupUsers")
+  adminGroups            UserGroup[]              @relation("GroupAdminUsers")
```

Comment:
what's this for?

Reply:
The 2 different lists are not really intentional, more an artefact of the db structure. It is to differentiate the roles the user has in a group. Maybe it is better modelled with a GroupUserRole table linking the Users table and the Group table.

What do you think?

Reply:
The group already specifies which users are admins, so whether we need this distinction here depends on when/where/how we need access to this information.

I'd add a relationship table `UserGroupMembership` which also specifies the user's role in the group.

Reply:
I agree with replacing these relationships to `UserGroupMembership`

---

> The group already has a list specifying which users are admins, so whether we need this distinction between groups and adminGroups here depends on when/where/how we need access to this information.

These are the other end of the many-to-many relationship so `UserGroup. adminUsers` maps to `User. adminGroups`.

Reply:
Done